sound/soc/sof/ipc4-topology.c

```diff
@@ -1748,11 +1788,11 @@ static const struct sof_ipc_tplg_widget_ops tplg_ipc4_widget_ops[SND_SOC_DAPM_TY
 	[snd_soc_dapm_scheduler] = {sof_ipc4_widget_setup_comp_pipeline, sof_ipc4_widget_free_comp,
 				    pipeline_token_list, ARRAY_SIZE(pipeline_token_list), NULL,
 				    NULL, NULL},
-	[snd_soc_dapm_pga] = {sof_ipc4_widget_setup_comp_pga, sof_ipc4_widget_free_comp,
+	[snd_soc_dapm_pga] = {sof_ipc4_widget_setup_comp_pga, sof_ipc4_widget_free_comp_pga,
 			      pga_token_list, ARRAY_SIZE(pga_token_list), NULL,
 			      sof_ipc4_prepare_gain_module,
 			      sof_ipc4_unprepare_generic_module},
-	[snd_soc_dapm_mixer] = {sof_ipc4_widget_setup_comp_mixer, sof_ipc4_widget_free_comp,
+	[snd_soc_dapm_mixer] = {sof_ipc4_widget_setup_comp_mixer, sof_ipc4_widget_free_comp_mixer,
```

Comment:
you should revisit sof_ipc4_widfet_free_comp_dai() now, wondering if you can use helpers introduced here?

Reply:
> you should revisit sof_ipc4_widfet_free_comp_dai() now, wondering if you can use helpers introduced here?

@plbossart Do you mean to refine the sof_ipc4_widget_free_comp_dai() to make it easier?

Reply:
I meant try to use the helpers rather than having a long list of kfree that seem to do the same thing as the helpers.